Core/include/Acts/Utilities/Grid.hpp

```diff
@@ -39,6 +40,30 @@ class IGrid {
   /// Get a dynamically sized vector of axis objects for inspection
   /// @return a vector of axis pointers
   virtual boost::container::small_vector<const IAxis*, 3> axes() const = 0;
+
+  /// Helper to print out the grid
+  /// @param os the output stream
+  /// @param grid the grid to print
+  /// @return the output stream
+  friend std::ostream& operator<<(std::ostream& os, const IGrid& grid) {
+    grid.toStream(os);
+    return os;
+  }
+
+  friend bool operator==(const IGrid& lhs, const IGrid& rhs) {
+    auto lhsAxes = lhs.axes();
+    auto rhsAxes = rhs.axes();
+    return lhsAxes.size() == rhsAxes.size() &&
+           std::equal(lhsAxes.begin(), lhsAxes.end(), rhsAxes.begin(),
+                      [](const IAxis* a, const IAxis* b) { return *a == *b; });
+  }
+
+  friend bool operator!=(const IGrid& lhs, const IGrid& rhs) {
+    return !(lhs == rhs);
```

Comment:
I guess this is not hot code so doesn't matter - the inequality operator could be "faster" by doing the inverse checks from above because it could evaluate faster to `true`

Reply:
Exactly, this is the slow path in any case (all of `IAxis` and `IGrid` should stay out of anything even remotely hot). 

I was thinking ease of implementation over performance.